### PR TITLE
Various improvements of mostly the files_editline.c code

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please. Let's see if it doesn't break something.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8368/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8368/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8368/

**Comment by vpodzime** [comment]
@cf-bottom let's go to jenkins again!

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8379/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8379/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8379/

**Comment by vpodzime** [comment]
> Alright, I triggered a build:
> 
> [Build Status](https://ci.cfengine.com//job/pr-pipeline/8379/)

Only SUSE 15 failed (unrelated).

**Review by larsewi — COMMENTED** [review]
LGTM, minor comments

File: cf-agent/files_editline.c [hunk 1]
```diff
@@ -2178,7 +2229,7 @@ static bool EditLineByColumn(EvalContext *ctx, Rlist **columns, const Attributes
     {
         /* internal separator, single char so split again */
 
-        if (strstr(RlistScalarValue(rp), a->column.column_value) || strcmp(RlistScalarValue(rp), a->column.column_value) != 0)
+        if (strstr(RlistScalarValue(rp), a->column.column_value) || !StringEqual(RlistScalarValue(rp), a->column.column_value))
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply cf-agent/files_editline.c:2232]: Are you sure we can simply change the condition here? I didn't check the code here, I just made some pattern-replace changes in these commits.

[the remark this answers is not in the document]
Comment (larsewi) [reply cf-agent/files_editline.c:2232]: Not sure. In my opinion you can just leave it as is :)

**Review by larsewi — APPROVED** [review]
ACK

File: cf-agent/files_editline.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (vpodzime) [reply cf-agent/files_editline.c:2232]: > [Lars] [2 minutes ago]
> IsStrIn looks for a string in a list of strings

**Comment by vpodzime** [comment]
@cf-bottom another round in jenkins, please, just in case

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8392/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8392/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8392/

**Pull request merged**